### PR TITLE
Chapter 2 (lighting) of the tutorials

### PR DESCRIPTION
**Pull request by frederikja163**

### CODE REVIEW

File: learn/chapter2/1-colors.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+model = glm::scale(model, glm::vec3(0.2f)); 
+The resulting drawing code for the lamp should then look something like this:
+
+
+lampShader.use();
+// set the model, view and projection matrix uniforms
+...
+// draw the lamp object
+glBindVertexArray(lightVAO);
+glDrawArrays(GL_TRIANGLES, 0, 36);
```
Comment (Perksey) [inline learn/chapter2/1-colors.md]: C code

File: learn/chapter2/1-colors.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+The main purpose of the lamp cube is to show where the light comes from. We usually define a light source's position somewhere in the scene, but this is simply a position that has no visual meaning. To show the actual lamp we draw the lamp cube at the same location of the light source. This is accomplished by drawing the lamp object with the lamp shader, ensuring the lamp cube always stays white, regardless of the light conditions of the scene.
+
+So let's declare a global vec3 variable that represents the light source's location in world-space coordinates:
+
+
+glm::vec3 lightPos(1.2f, 1.0f, 2.0f);
+We then want to translate the lamp's cube to the light source's position before drawing it and we'll also scale it down a bit to make sure the lamp isn't too dominant:
+
+
+model = glm::mat4(1.0f);
```
Comment (Perksey) [inline learn/chapter2/1-colors.md]: glm

File: learn/chapter2/1-colors.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+unsigned int lightVAO;
+glGenVertexArrays(1, &lightVAO);
+glBindVertexArray(lightVAO);
+// we only need to bind to the VBO, the container's VBO's data already contains the correct data.
+glBindBuffer(GL_ARRAY_BUFFER, VBO);
+// set the vertex attributes (only position data for our lamp)
+glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), (void*)0);
+glEnableVertexAttribArray(0);
+The code should be relatively straightforward. Now that we created both the container and the lamp cube there is one thing left to define and that is the fragment shader:
+
```
Comment (Perksey) [inline learn/chapter2/1-colors.md]: Upon reviewing this I realised that this is just a copy + paste. Haha no matter, keep it up!

Comment (frederikja163) [reply learn/chapter2/1-colors.md]: Haha, yeah i just wanted to move from my laptop - desktop

**Pull request merged**